memory/provider.go

```diff
@@ -103,17 +103,19 @@ func (mp *Provider) Count() int {
 
 // NeedGC need gc
 func (mp *Provider) NeedGC() bool {
-	if mp.expiration == 0 {
-		return false
-	}
-
 	return true
 }
 
 // GC session garbage collection
 func (mp *Provider) GC() {
 	for _, kv := range mp.memoryDB.D {
-		if time.Now().Unix() >= (kv.Value.(*Store).lastActiveTime + mp.expiration) {
+		store := kv.Value.(*Store)
+		// Do not expire the session if expiration is set to 0
+		if store.GetExpiration() == 0 {
```

Comment:
Save value in new variable, and use it, because expiration could be change by another gorutine.

Reply:
done